File: clubcontest/src/app/models/flight.ts
import { IUser } from './user';
import { ILoggerPoint } from './flight';
import 'leaflet';

export interface IFlight {
    Id: number;
    DayId: number;
    IgcFile: string;
    Points: ILoggerPoint[];
    UserId: number;
    User: IUser;

    TurnPoints?: string,
    turnPoints: ILoggerPoint[],
    Distance?: number;
    Duration?: number;
    Speed?: number;
    Finished?: boolean;  
    Score:number;
    GliderType:string;
    Handicap:number;
    PilotName:string;
    
    Layer: L.Layer;
    IsVisible: boolean;
    isExpanded:boolean;
    isLoading:boolean;
    isSelected:boolean;
    color:string;


}

export interface ILoggerPoint {
    Latitude: number,
    Longitude: number,
    Time?: Date;
}


export interface IFlightScoring {
    points: ILoggerPoint[],
    totalDistance: number;
    time: number;
    speed: number;
    finished: boolean;
}

File: ClubContestMvcApp/DataModels/Flight.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Newtonsoft.Json;
using SampleMvcApp.Extensions;

namespace SampleMvcApp.DataModels
{
    public class Flight
    {
		public int Id { get; set; }
		public int DayId { get; set; }

		[JsonIgnore]
		public string IgcFile { get; set; }

		public int UserId { get; set; }
		
		public User User { get; set; }

		public Day Day { get; set; }

	  public Point[] Points { get; set; }

		public double? Speed { get; set; }

		public double? Distance { get; set; }

		public double? Duration { get; set; }

		public bool? Finished { get; set; }

		public string TurnPoints { get; set; }
		
		public string GliderType { get; set; }

		public double Handicap { get; set; }

	  public string PilotName {get;set;}
	}
}
